io-classes/src/Control/Monad/Class/MonadSTM/Internal.hs

```diff
@@ -535,6 +544,51 @@ class MonadInspectSTM m
                       -> m ()
   traceTSemIO = \v f -> atomically (traceTSem Proxy v f)
 
+traceTVarShow :: (MonadTraceSTM m, Show a)
+               => proxy m
+               -> TVar m a
+               -> STM m ()
+traceTVarShow p tvar =
+  traceTVar p tvar (\pv v -> pure $ TraceString $ case (pv, v) of
+          (Nothing, st') -> "Created: " <> show st'
```

Comment:
This will never show up in the trace because when one calls `traceTVarShow`, the `tvar` already exists.

Reply:
I suspected that, but still the type of the function made me pattern match on that. Should I throw an error instead?

Reply:
Done